libkernelflinger/vars.c

```diff
@@ -254,32 +254,70 @@ static void set_provisioning_mode(BOOLEAN provisioning)
 	current_state = provisioning ? UNLOCKED : LOCKED;
 }
 
-enum device_state get_current_state()
+static EFI_STATUS read_efi_device_state(UINT8 *state)
 {
-	UINT8 *stored_state;
-	UINTN dsize;
 	EFI_STATUS ret;
+	UINT8 *stored_state = NULL;
+	UINTN dsize;
 	UINT32 flags;
+
+	ret = get_efi_variable((EFI_GUID *)&fastboot_guid, OEM_LOCK,
+				       &dsize, (void **)&stored_state, &flags);
+	if (ret == EFI_NOT_FOUND) {
+		*state = UNKNOWN_STATE;
+		ret = EFI_SUCCESS;
+		goto out;
+	}
+	if (EFI_ERROR(ret)) {
+		// Read failed
+		if (ret == EFI_NOT_FOUND)
+			log(L"Can not find device state EFI variable");
+		else
+			efi_perror(ret, L"Failed to read device state from EFI variable");
+		goto out;
```

Comment:
Same here you could just return since no buffer has been allocated.

Reply:
Just make more stable, need not care the detail of get_efi_variable.